gateway-ha/src/test/java/io/trino/gateway/ha/router/TestStochasticRoutingManager.java

```diff
@@ -58,15 +66,77 @@ public void addMockBackends()
             haRoutingManager.updateBackEndHealth(backend, true);
         }
 
-        //Keep only 1st backend as healthy, mark all the others as unhealthy
-        assertThat(backendManager.getAllActiveBackends()).isNotEmpty();
+        // adhoc group
+        // add mock backend
+        ProxyBackendConfiguration proxyBackend = new ProxyBackendConfiguration();
+        proxyBackend.setActive(true);
+        proxyBackend.setRoutingGroup(ADHOC);
+        proxyBackend.setName(ADHOC);
+        proxyBackend.setProxyTo(ADHOC + ".trino.example.com");
+        proxyBackend.setExternalUrl("trino.example.com");
+        backendManager.addBackend(proxyBackend);
+        //set backend as healthy start with
+        haRoutingManager.updateBackEndHealth(ADHOC, true);
+    }
 
-        for (int i = 1; i < numBackends; i++) {
-            backend = groupName + i;
+    @BeforeEach
+    public void init()
+    {
+        headers.clear();
+        // before each class, reset health status for all as true
+        haRoutingManager.updateBackEndHealth(ADHOC, true);
+        for (int i = 0; i < numTestBackends; i++) {
+            String backend = groupName + i;
+            haRoutingManager.updateBackEndHealth(backend, true);
+        }
+    }
+
+    @Test
+    public void testActiveBackendCount()
+    {
+        assertThat((long) backendManager.getAllActiveBackends().size()).isEqualTo(numTestBackends + 1);
+    }
+
+    @Test
+    public void testNoHealthyTestBackend()
```

Comment:
I don't know what's the correct response for:
* missing a user header
* all backends are not healthy

I'm okay with adding these tests since they reflect the current behavior.
We might want to review them later.

Reply:
Well this test is to test if it routes to adhoc on case
where all request group (test_group) backends are not healthy